### ISSUE TITLE
Help

### ISSUE THREAD

**Issue by olliemay** [richardingham/octopus#2]
Hi Richard, 

I'm trying to find a simple way of connect a Gilson 402 duel channel syringe pump to a raspberry pi. Currently it is connect via a Gilson 508 GSIOC to RS232 converter connect to the RPi through a USB to RS232 converter. The idea is to build a custom flow chemistry rig using the pi as a controller which can also connect to in-line analytics (UV-vis, FTIR and NMR). 

I'm new to python so trying to get my head around how to deploy your code as you also include web options through twisted? Could you upload an idiots a step by step guide to how/where to install on a RPi or Mac and how to then set the various variables (flow rate, volumes etc). I have heard LabView may be alternative option but it looks like you've already creaked most of what i would like to do - i just need to understand how you did it and how to use your code!

I have looked at the GSIOC protocol and have tried to build my own PySerial python script based on someone else's code but haven't had much luck - any help would be fantastic!

Kind regards, 
Ollie

**Comment by andyfinity** [comment]
I'm not Richard, but I have been working on custom GSIOC code for a while now. It looks like this repo is abandoned at this point, so maybe I could give some pointers.

There's a bunch of pretty decent documentation in the doc folder. It'll give you some idea about how to set up and use Octopus. Note that only a few Gilson devices are supported (506C, 233XL, 402, and 151/155 series). You are able to install the repo wherever you like. I'm not entirely clear on what you're trying to do, but this repo may not provide a huge amount of utility to you. As a result, you may want to consider writing your own control software. If the devices you want aren't supported, you'll end up writing your own code anyway.

My experience with GSIOC is: it's pretty bad, but not terrible. There's a ton of overhead and it's very slow, but communication over RS232 is fairly reliable. I've had a lot of trouble using aftermarket GSIOC cables due to crosstalk. Look for the GSIOC technical manual to learn how to send commands, then look in the user manual for each device to find possible commands to send.

Consider using an Arduino or other embedded device to do the low-level communication. I've found that a PC or RPi (or anything that has to convert RS232 to USB) introduces a large amount of per-character delay (up to 10ms per round trip). An embedded device with tight coupling to the UART hardware will provide very quick character-by-character communication. The performance improvement will be substantial with long strings of data (like an analog FIFO).

Contact Gilson support. They've been pretty helpful for me. My biggest problem is decoding FIFO data, and Gilson seems reluctant to give me that information. Luckily, this repo shows how to decode it. I simply need to translate the relevant code from Python to C++ for my project.

*tl;dr: docs have the info you need, and consider writing your own code.*
